Framework/API/src/LogManager.cpp

```diff
@@ -176,21 +179,74 @@ const Types::Core::DateAndTime LogManager::endTime() const {
     }
   }
 
-  if (hasProperty("proton_charge")) {
-    try {
-      Kernel::TimeSeriesProperty<double> *protonchargelog =
-          dynamic_cast<Kernel::TimeSeriesProperty<double> *>(getProperty("proton_charge"));
-      return DateAndTime(protonchargelog->lastTime());
-    } catch (std::invalid_argument &) { /*Swallow and move on*/
-    }
+  try {
+    return getLastPulseTime();
+  } catch (Exception::NotFoundError &) { /*Swallow and move on*/
+  } catch (std::invalid_argument &) {    /*Swallow and move on*/
   }
 
   throw std::runtime_error("No valid end time has been set for this run.");
 }
 
-bool LogManager::hasStartTime() const { return hasProperty(START_TIME_NAME) || hasProperty("run_start"); }
+bool LogManager::hasStartTime() const {
+  try {
+    auto temp = startTime();
```

Comment:
I think the compiler will let you combine this into
```c++
UNUSED_ARG(startTime());
```

Reply:
That is no longer a concern as I have changed both methods to not use try/catch.